app/mailers/application_mailer.rb

```diff
@@ -1,7 +1,17 @@
+require 'uri'
+
 class ApplicationMailer < ActionMailer::Base
 
   default :from => Setting[:email_reply_address] || "noreply@foreman.example.org"
 
+  def mail(headers = {}, &block)
+    return if headers.blank?
```

Comment:
@ohadlevy  ActionMailer::Base tests whether headers are empty, so I was trying to avoid the case where headers were never blank because of my changes...but anyway, the latest update I think is less intrusive into the mail method  and addresses your concern here